src/nbl/core/alloc/refctd_memory_resource.cpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (C) 2018-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#include "nbl/core/alloc/refctd_memory_resource.h"
+
+using namespace nbl;
+using namespace core;
+
+static smart_refctd_ptr<refctd_memory_resource> null_memory_resource = nullptr;
```

Comment:
the null memory resource can go live inside the `getNullMemoryResource` (statics inside functions work exactly the way you think)